### PR TITLE
AKI-699 FVM callback delegate call fix

### PR DESCRIPTION
**Pull request by AionJayT**
## Description

-

Fixes Issue # .

## Type of change

- [ ] Bug fix.
- [ ] New feature.
- [ ] Enhancement.
- [ ] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

-

### CODE REVIEW

File: modAionImpl/src/org/aion/zero/impl/forks/ForkUtility.java [hunk 1]
```diff
@@ -54,6 +54,16 @@ public ForkUtility(Properties properties, Logger log) {
 
             enableNonceFork(maybeNonceFork.get());
         }
+
+        Optional<Long> maybeSignatureSwapFork = loadSignatureSwapForkNumberFromConfig(properties);
+        if (maybeSignatureSwapFork.isPresent()) {
+            if (maybeSignatureSwapFork.get() < 2) {   // AKI-419, Constrain the minimum unity fork number
+                log.warn("The nonce fork number cannot be less than 2, set the fork number to 2");
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/forks/ForkUtility.java:61]: the warning should say fork1.7; also the comment regarding the AKI can be removed

File: modAionImpl/src/org/aion/zero/impl/forks/ForkUtility.java [hunk 2]
```diff
@@ -130,6 +140,32 @@ public void enableNonceFork(long nonceForkBlockHeight) {
...
+
+    /**
+     * Returns a boolean value indicating if the signature swap fork is active for the given context (block
+     * number). We want the fork block itself to be a PoW block subject to the old pre-Unity rules,
+     * so we use a strict greater than comparison.
+     *
+     * @return {@code true} if the signature swap fork fork is active for the given context (block number), {@code
+     *     false} otherwise
+     */
+    public boolean isSignatureSwapForkActive(long contextBlockNumber) {
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/forks/ForkUtility.java:155]: Depending on the use case the second method checking equality may not be necessary and the first could check >=. Do we have a clear specification on when this fork becomes active?

File: modAionImpl/src/org/aion/zero/impl/forks/ForkUtility.java [hunk 3]
```diff
@@ -200,4 +236,19 @@ public long getNonceForkBlockHeight() {
...
     }
+
+    /**
+     * Determine fork 1.7 fork number from Aion Config.
+     *
+     * @return 1.7 fork number, if configured; {@link Optional#empty()} otherwise.
+     * @throws NumberFormatException if "fork1.7" present in the config, but not parseable
+     */
+    private static Optional<Long> loadSignatureSwapForkNumberFromConfig(Properties properties) {
+        String nonceFork = properties.getProperty("fork1.7");
```
Comment (AlexandraRoatis) [inline modAionImpl/src/org/aion/zero/impl/forks/ForkUtility.java:247]: rename `nonceFork` to `fork`

File: modAionImpl/build.gradle [hunk 4]
```diff
@@ -54,7 +54,7 @@ dependencies {
     compile project(':modEvtMgrImpl')
     compile project(':modPrecompiled')
     compile project(':modTxPool')
-    compile files("${rootProject.projectDir}/lib/fastvm-4d55b49.jar")
+    compile files("${rootProject.projectDir}/lib/fastvm-9f1f780.jar")
```
Comment (AionJayT) [inline modAionImpl/build.gradle:57]: add commit link into the commit message

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java [hunk 5]
```diff
@@ -132,6 +132,7 @@ public long estimateTxNrg(AionTransaction tx, Block block) {
             boolean fork040enabled = false;
             boolean checkBlockEnergyLimit = false;
             boolean unityForkEnabled = false;
+            boolean signatureSwapForkEnabled = ((AionBlockchainImpl)aionHub.getBlockchain()).forkUtility.isSignatureSwapForkActive(block.getNumber());
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:135]: add get forkUtil method in the aionhub

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java [hunk 6]
```diff
@@ -132,6 +132,7 @@ public long estimateTxNrg(AionTransaction tx, Block block) {
             boolean fork040enabled = false;
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:132]: need to check the fork

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java [hunk 7]
```diff
@@ -172,6 +174,7 @@ public AionTxReceipt callConstant(AionTransaction tx, Block block) {
...
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:174]: add a fix for fork040 flag

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java [hunk 8]
```diff
@@ -172,6 +174,7 @@ public AionTxReceipt callConstant(AionTransaction tx, Block block) {
...
             boolean checkBlockEnergyLimit = false;
             boolean unityForkEnabled = false;
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:176]: fix the check

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java [hunk 9]
```diff
@@ -132,6 +132,7 @@ public long estimateTxNrg(AionTransaction tx, Block block) {
...
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:134]: fix the check

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java (hunk 6 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:132]: AKI-702

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java (hunk 7 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:174]: AKI-702

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java (hunk 8 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:176]: AKI-702

File: modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java (hunk 9 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java:134]: AKI-702

**Pull request merged**